Neos.Flow/Classes/Persistence/Doctrine/ObjectValidationAndDeDuplicationListener.php

```diff
@@ -89,20 +90,21 @@ private function deduplicateValueObjectInsertions()
         $entityInsertions = $unitOfWork->getScheduledEntityInsertions();
 
         $knownValueObjects = [];
-        foreach ($entityInsertions as $entity) {
+        foreach ($entityInsertions as $oid => $entity) {
             $className = TypeHandling::getTypeForValue($entity);
             $classSchema = $this->reflectionService->getClassSchema($className);
             if ($classSchema !== null && $classSchema->getModelType() === ClassSchema::MODELTYPE_VALUEOBJECT) {
                 $identifier = $this->persistenceManager->getIdentifierByObject($entity);
 
                 if (isset($knownValueObjects[$className][$identifier]) || $unitOfWork->getEntityPersister($className)->exists($entity)) {
-                    $unitOfWork->scheduleForDelete($entity);
+                    unset($entityInsertions[$oid]);
                     continue;
                 }
 
                 $knownValueObjects[$className][$identifier] = true;
             }
         }
+        ObjectAccess::setProperty($unitOfWork, 'entityInsertions', $entityInsertions, true);
```

Comment:
scary shit, but seems there is no better way :/

Reply:
It would be great, if we could modify the result of a constructor. So that `new ValueObject()` returns a managed entity, if there already is a fitting ValueObject in the database. But I found no way for a stunt like that.

Probably one could try to adjust the ProxyGenerator to replace construction of ValueObjects with some service glue... but thats definitely no low hanging fruit.

The object deduplication is in its current way a scary hack... even `scheduleForDelete` does not really change that. It is public, but marked as internal - so we are definitely doing something "unsupported" here.

Reply:
As long as the scary shit is not replaced by even scarier shit… 🤷‍♂️